framework/contracts/account/src/modules.rs

```diff
@@ -297,52 +294,41 @@ fn configure_adapter(
     Ok(wasm_execute(adapter_address, &adapter_msg, vec![])?.into())
 }
 
-/// Add a contract to the whitelist
-pub(crate) fn _whitelist_modules(deps: DepsMut, module_addresses: Vec<Addr>) -> AccountResult<()> {
-    let mut whitelisted_modules = WHITELISTED_MODULES.load(deps.storage)?;
-
+/// Update whitelist by adding or removing addresses
+pub(crate) fn _update_whitelisted_modules(
+    storage: &mut dyn Storage,
+    to_add_module_addresses: Vec<Addr>,
+    to_remove_module_addresses: Vec<Addr>,
+) -> AccountResult<()> {
+    let mut whitelisted_modules = WHITELISTED_MODULES.load(storage)?.0;
+    let new_len = (whitelisted_modules.len() + to_add_module_addresses.len())
+        .checked_sub(to_remove_module_addresses.len())
+        // If overflowed - tried to remove not whitelisted
+        .ok_or(AccountError::NotWhitelisted {})?;
     // This is a limit to prevent potentially running out of gas when doing lookups on the modules list
-    if whitelisted_modules.0.len() >= LIST_SIZE_LIMIT {
+    if new_len > WHITELIST_SIZE_LIMIT {
         return Err(AccountError::ModuleLimitReached {});
     }
 
-    for module_addr in module_addresses.into_iter() {
-        if whitelisted_modules.0.contains(&module_addr) {
+    for module_addr in to_add_module_addresses {
+        if whitelisted_modules.contains(&module_addr) {
             return Err(AccountError::AlreadyWhitelisted(module_addr.into()));
         }
 
         // Add contract to whitelist.
-        whitelisted_modules.0.push(module_addr);
+        whitelisted_modules.push(module_addr);
     }
 
-    WHITELISTED_MODULES.save(deps.storage, &whitelisted_modules)?;
-
-    Ok(())
-}
-
-/// Remove a contract from the whitelist
-pub(crate) fn _remove_whitelist_modules(
-    deps: DepsMut,
-    addresses_to_remove: Vec<Addr>,
-) -> AccountResult<()> {
-    let mut len: i8 = addresses_to_remove.len() as i8;
-
-    WHITELISTED_MODULES.update(deps.storage, |mut whitelisted_modules| {
-        whitelisted_modules.0.retain(|addr| {
-            // retain any addresses that are not in the list of addresses to remove
-            if addresses_to_remove.contains(addr) {
-                len -= 1;
-                false
-            } else {
-                true
-            }
-        });
-        Ok::<_, StdError>(whitelisted_modules)
-    })?;
+    whitelisted_modules.retain(|addr| {
+        // retain any addresses that are not in the list of addresses to remove
+        !to_remove_module_addresses.contains(addr)
+    });
```

Comment:
Sometimes we do add+remove, for example on UpdateWhitelist action and Upgrade if that's adapter. It was inconsistent when we add or remove first, also takes 2 loads and saves, decided to merge it instead

Reply:
What happens if an address is both in to-add and to-remove? Should be clarified in a doc-comment.